patterns/state/ships/src/main/java/com/scottlogic/dp/state/ships/DriftingState.java

```diff
@@ -2,30 +2,50 @@
 
 class DriftingState extends ShipState {
     @Override
-    public void startFishing(FishingShip ship) {
+    public ShipState startFishing() {
         System.out.println("Have to anchor first!");
+        return this;
     }
 
     @Override
-    public void dropAnchor(FishingShip ship) {
+    ShipState stopFishing() {
+        System.out.println("Haven't thrown that net yet, mate.");
+        return this;
+    }
+
+    @Override
+    ShipState dropAnchor() {
         System.out.println("Lowering the anchor! And shields.");
-        ship.setState(new AnchoredState());
+        return new AnchoredState();
     }
 
     @Override
-    public void raiseAnchor(FishingShip ship) {
+    ShipState raiseAnchor() {
         System.out.println("The anchor is not set");
+        return this;
     }
 
     @Override
-    public void startSailing(FishingShip ship) {
+    ShipState startSailing(Engine engine) {
         System.out.println("Sailing!");
-        ship.setState(new SailingState());
+        engine.start();
+        if (engine.getEngineState() == EngineState.BROKEN) {
```

Comment:
If I had another state that would render the Ship Stranded (e.g. outOfFuel), how would this condition look like?

Reply:
Could the outOfFuel state happen theoretically happen at any time? or after you ask the ship to sail()?

Reply:
Good question. That aside, what I wanted to point out that this condition could be encapsulated in some sort of .isOperational() method, so if this condition is used elsewhere and should the condition change you won't have to go everywhere and change it. Additionally, the state of the engine will be hidden :)
Am I making any sense at all?

Reply:
That makes sense, I forgot what we spoke about last time, I'll have a look